### ISSUE TITLE
Publish to NuGet

### ISSUE THREAD

**Issue by djdembeck** [rmcrackan/AudibleApi#18]
Hello and thanks for this project :)

I'm building a project and would like to use this source instead of using the Python one. However, it looks like there is no NuGet package support yet. I think this would be greatly helpful for people wishing to use this, and should be relatively simple through Github Actions.

**Comment by rmcrackan** [comment]
Good idea. This should definitely happen eventually. It's just that there are so many things above it in my priority list. I've never set anything up before to deploy/publish to nuget. If you have some resources which would make it easier, then I'm more likely to push it ahead in the queue. It would be great to be able to offer this as a nuget pkg

I started to type out a reply here about how all to use it. This is something I really need to provide more publicly (which is to say: I really ought to have done this by now) though so I'll create a 'getting started' section on the main page instead. Expect this later this evening.

**Comment by rmcrackan** [comment]
https://github.com/rmcrackan/AudibleApi#getting-started

It's not much, but's it's a start. I can add more when I see what real-world questions come up.

**Comment by djdembeck** [comment]
Thanks so much for writing that up!

I'm still green on C#, but this seems like an easy task to accomplish. The difficulty comes in with `Dinah.Core`. This is likely another package you will want to publish, then include as a dependency here. I could quite easily publish the former, but AudibleApi will need some changes I am not sure how you want to approach, in order to streamline the dependency.

As it stands, there's also no mention of Dinah.Core in the readme, but I believe it needs to be cloned by any user, alongside of this repo.

Again, since I am not sure the restructuring you would like to do, I haven't submitted a PR with these Github Actions, but they are both the same (just changing the name). All you need is a Secret setup as NUGET_API_KEY with your Nuget account API key.

Dinah.Core Action example: https://github.com/djdembeck/Dinah.Core/blob/master/.github/workflows/main.yml

Dina.Core on Nuget example (I will delete it if/when you get it setup under your account):
https://www.nuget.org/packages/Dinah.Core/

Github Action template used:
https://github.com/marketplace/actions/publish-nuget

**Comment by rmcrackan** [comment]
Please see if you can remove your Dinah.Core from nuget. I really hope it allows me to reuse the name.

In the devops world, I am firmly dev with pretty much no ops so this is all greek to me. Here's my script so far. What am I missing?

sign up with a microsoft account. necessary for nuget  
login to nuget.org  
nuget.org > username (in upper right) > API Keys  
create  
copy

in my code space:  
manually create yaml in repo  
\.github\workflows\publish.yml  
example yaml template for publishing to nuget: https://github.com/marketplace/actions/publish-nuget  
push to github

github.com > settings > secrets  
click 'New repository secret'  
name must match ${{secrets.NUGET_API_KEY}} in .yml file  
Name: NUGET_API_KEY  
Value: paste api key from above  
click 'Add secret'

**Comment by djdembeck** [comment]
Yes, you have the steps down perfectly! And then uncomment:
- `PROJECT_FILE_PATH` and change the path.
- `PACKAGE_NAME` and change the name
- `VERSION_STATIC` and change the version (I think Nuget starts at 1.0.0 no matter what you set).
- `NUGET_KEY`

I added you as an owner of the nugget repo, so you can take ownership and remove me!

If you need anything else from me I'm happy to help. Can't wait to start using your project!

**Comment by rmcrackan** [comment]
Thanks so much!

I'd like to do the same for other projects. How do I initially create a package? When I go to [username] > Manage Packages > Add new, it wants me to upload. I have nothing to upload yet.
